---
layout: page_single
title: Thanks for checking out my poster!
permalink: /anaerobe/
comments: false
image:
  feature: anaerobe2016
---

My work leverages metagenomics, metatranscriptomics, and metabolic modeling to identify the nutrient of *Clostridium difficile* in the gut during infection.

For more about me or my project, take a look at my [website](http://mjenior.github.io)!

Methods
---------------------

Differential plating

Toxin titer 

High-throughpt sequencing

Bipartite metabolic modeling

16S analysis

Metagenomic-enabled Metatranscriptomics



References
---------------------

1. Lessa FC et. al. (2015). Burden of *Clostridium difficile* Infection in the United States. *N Engl J Med*. vol. 372; 825-834.

2. Theriot CM, Koenigsknecht MJ, Carlson PE, Hatton GE, Nelson AM, Li B, & Young VB. (2014). Antibiotic-induced shifts in the mouse gut microbiome and metabolome increase susceptibility to *Clostridium difficile* infection. *Nature Comm*. Vol. 5: 3114.

3. Wilson KH & Perini F. (1988). Role of competition for nutrients in suppression of *Clostridium difficile* by the colonic microflora. *Inf & Imm*. Vol. 56 (10); 2610-2614.

4. Patil KR & Nielsen J. (2005). Uncovering transcriptional regulation of metabolism by using metabolic network topology. *PNAS*. vol. 102 (8); 2685â€“2689.

5. Baker BJ, Sheik CS, Taylor CA, Jain S, Bhasi A, Cavalcoli JD, & Dick GJ. (2012). Community transcriptomic assembly reveals microbes that contribute to deep-sea carbon and nitrogen cycling. *ISME*. Vol. 7 (10); 1962-1973.

