---
layout: page_single
title: Thanks for checking out my poster!
permalink: /anaerobe/
comments: false
image:
  feature: anaerobe2016.jpg
---

My name is [Matt Jenior](https://www.linkedin.com/in/matthew-jenior-61466662).  The work I do in the [Schloss lab](http://www.schlosslab.org/) leverages metagenomics, metatranscriptomics, and metabolic modeling to identify the nutrient niche of *Clostridium difficile* in the gut during infection.  With this information we can potentially begin to design targeted probiotics to eliminate/prevent colonization of *C. difficile*.  Below is some supplementary info for my poster, it will give you some background on how I generated the data and where I got some of my ideas.  For more about me or my science, take a look at my [website](http://mjenior.github.io)!

---

Methods (Abbreviated)
---------------------

**Differential plating:**

All done anaerobically*

Vegetative cells = plated on CCFA (no taurocholate)

Spores = 60 degrees Celsius for 30 minutes, then plated on TCCFA


**Toxin titer:**

Performed using the standard Vero cell rounding assay by [Jhansi Leslie](https://www.researchgate.net/profile/Jhansi_Leslie) in [Vince Young's lab](https://sites.google.com/a/umich.edu/younglab/).


**High-throughpt sequencing:**

Illumina MiSeq - 16S rRNA gene (2 x 250 bp)

Illumina HiSeq - metagenomes (2 x 250 bp) and metatranscriptomes (2 x 50 bp)

**16S rRNA gene analysis:**

[mothur](http://www.mothur.org/wiki/MiSeq_SOP) of course!


**Bipartite metabolic modeling:**

I developed a novel algorithm for calculating the importance of biochemical reaction substrates in an organism's metabolic network.  These valuse are based on the transcription of the adjacent enzymes that act on or produce the given substrate.  The resulting score reflects the liklihood that a given compound is acquired from the environment.


**Metagenomic-enabled Metatranscriptomics:**

Metagenomes were assembled and gene content was then annotated.  Metatranscriptomic read mapping was then normalized to metagenomic coverage before comparing between conditions.


---

References
---------------------

1. Lessa FC, et. al. (2015). Burden of *Clostridium difficile* Infection in the United States. *N Engl J Med*. vol. 372; 825-834.

2. Theriot CM, Koenigsknecht MJ, Carlson PE, Hatton GE, Nelson AM, Li B, & Young VB. (2014). Antibiotic-induced shifts in the mouse gut microbiome and metabolome increase susceptibility to *Clostridium difficile* infection. *Nature Comm*. Vol. 5: 3114.

3. Wilson KH & Perini F. (1988). Role of competition for nutrients in suppression of *Clostridium difficile* by the colonic microflora. *Inf & Imm*. vol. 56 (10); 2610-2614.

4. Patil KR & Nielsen J. (2005). Uncovering transcriptional regulation of metabolism by using metabolic network topology. *PNAS*. vol. 102 (8); 2685â€“2689.

5. Schubert AM, Sanini H, S& Schloss PD. (2015). Antibiotic-Induced Alterations of the Murine Gut Microbiota and Subsequent Effects on Colonization Resistance against Clostridium difficile. *MBio*. vol. 6 (4); e00974-15.

6. Baker BJ, Sheik CS, Taylor CA, Jain S, Bhasi A, Cavalcoli JD, & Dick GJ. (2012). Community transcriptomic assembly reveals microbes that contribute to deep-sea carbon and nitrogen cycling. *ISME*. vol. 7 (10); 1962-1973.

---

I would love to talk to you about science!  Find me on the [Anaerobe 2016 Whova app](http://whova.com/portal/anaer_201607) so we can meet up!