---
layout: post
title:  "And now for something completely different: dN/dS"
date:   2014-11-17
comments: true
---

While I wait for the assemblies to merge I'm going to take a detour to another project I've been working on.

dN/dS measures the ratio of non-synonymous to synonymous substitutions in two or more genes. This ratio can be used to infer selection. Normally, to measure dN/dS you would use a reference genome of a closely related species to infer selection on your genome of interest. Unfortunately, with metagenomics we don't have that luxury so I'm going to do things a little differently. I'm not particularly interested in determining selection from an ancestral state, rather I'd like to know the differences in the same gene between datasets. This means I can't tell direction but that's alright, that's not what I'm interested in.

Doing this with metagenomic data is a little different, most importantly we don't know what genes we should be comparing at the outset. To determine what genes to align I did a blast of all full length genes that had a coverage >= 5x for each dataset against a pool of the other datasets. From this, I took the top hit from each dataset and aligned these genes. Each dataset served as a "reference" for the others in determining dN/dS.

Let's walk though one analysis.

Datasets:

* 35m
* 34f
* 10f
* 8m
* 6m
* 4m

In this example I will be using dataset 35m as the reference and comparing it to the others. The first step is to run a blastp against all the other datasets. Blast allows you to blast vs multiple databases making this step a bit easier.

After the blast was done I picked out the top hit to each dataset for each gene and got a nice list of common genes, one from each dataset.

~~~~
35m_32_gene_2   10f_994824_gene_1,34f_1126744_gene_4,4m_1018363_gene_3,6m_147874_gene_4,8m_1149_gene_7
35m_32_gene_4   10f_83189_gene_1,34f_140824_gene_3,4m_1025781_gene_2,6m_1045175_gene_2,8m_59057_gene_12
35m_32_gene_6   10f_1352005_gene_4,34f_201564_gene_5,4m_9097_gene_3,6m_241861_gene_2,8m_6591_gene_9
35m_32_gene_7   10f_992966_gene_2,34f_12937_gene_3,4m_9097_gene_4,6m_47942_gene_2,8m_6591_gene_10
35m_32_gene_8   10f_1352005_gene_6,34f_1150754_gene_7,4m_1279389_gene_3,6m_47942_gene_3,8m_6591_gene_11
35m_32_gene_9   10f_1245_gene_1,34f_1150754_gene_6,4m_9097_gene_6,6m_47942_gene_4,8m_6591_gene_12
35m_32_gene_10  10f_1245_gene_2,34f_1150754_gene_5,4m_207416_gene_2,6m_47942_gene_5,8m_6591_gene_13
35m_32_gene_12  10f_1245_gene_4,34f_1145947_gene_1,4m_996565_gene_2,6m_1118993_gene_1,8m_988255_gene_6
35m_32_gene_13  10f_1245_gene_5,34f_68810_gene_6,4m_9097_gene_10,6m_924686_gene_3,8m_94497_gene_13
35m_32_gene_14  10f_1245_gene_6,34f_68810_gene_5,4m_9097_gene_11,6m_924686_gene_4,8m_94497_gene_14
35m_32_gene_15  10f_1245_gene_7,34f_68937_gene_4,4m_9097_gene_12,6m_1126902_gene_3,8m_967727_gene_3
35m_32_gene_16  10f_1312177_gene_4,34f_1335560_gene_11,4m_9097_gene_13,6m_913_gene_5,8m_1211498_gene_7
35m_32_gene_17  10f_1368769_gene_3,34f_20316_gene_6,4m_9097_gene_14,6m_913_gene_6,8m_965529_gene_6
35m_32_gene_18  10f_1245_gene_10,34f_20316_gene_5,4m_9097_gene_15,6m_1015904_gene_4,8m_965529_gene_5
35m_32_gene_19  10f_1245_gene_11,34f_20316_gene_4,4m_9097_gene_16,6m_1015904_gene_5,8m_965529_gene_4
35m_32_gene_20  10f_1245_gene_12,34f_20316_gene_3,4m_9097_gene_17,6m_1015904_gene_6,8m_965529_gene_3
35m_32_gene_21  10f_1245_gene_13,34f_20316_gene_2,4m_993839_gene_2,6m_1015904_gene_7,8m_965529_gene_2
35m_32_gene_22  10f_1245_gene_14,34f_20316_gene_1,4m_3015_gene_2,6m_1015904_gene_8,8m_965529_gene_1
35m_32_gene_23  10f_37_gene_4,34f_1132837_gene_1,4m_3015_gene_3,6m_950550_gene_4,8m_1122005_gene_4
35m_32_gene_24  10f_37_gene_3,34f_1124987_gene_3,4m_1079284_gene_2,6m_1149866_gene_3,8m_991984_gene_2
35m_32_gene_25  10f_37_gene_2,34f_1124987_gene_4,4m_2430_gene_2,6m_1149866_gene_4,8m_991984_gene_3

~~~~

The first column is the reference and the names break down like this: dataset_contigNumber_gene_geneNumber.

This allows us to make a few observations by eye. All of these genes are from the same contig, 32, but hit a few different contigs in each dataset. This contig 32 from the 35m dataset is probably similar to the 1245 contig in the 10f dataset and similarly the 20316 contig in 34f, 9097 in 4m, 1015904 in 6m and 965529 in 8m. At first it might seem strange that not all of the hits to, say, the 10f dataset are on the same contig, but like most things, strange is a good place to start. These genes were pulled as top hits by e-value from the blast output so it seems like there was a better match to 35m_32_gene_16 on the 10f contig 1312177. This might mean the 10f contig 1245 is missing this gene or it's in a different position or it's really different or something else entirely. To investigate further we can go back to the gene calling data and look at the coordinates of the gene on the contig. If this sounds exciting (and it should!) stay tuned for the next installment of the case of the missing gene!
